copy/single.go

```diff
@@ -365,26 +385,34 @@ func (ic *imageCopier) compareImageDestinationManifestEqual(ctx context.Context,
 		return nil, nil
 	}
 
+	compressionAlgos := []compressiontypes.Algorithm{}
+	for _, srcInfo := range ic.src.LayerInfos() {
+		compression := ic.getCompressionFromBlob(srcInfo)
+		compressionAlgos = append(compressionAlgos, *compression)
+	}
+
 	// Destination and source manifests, types and digests should all be equivalent
 	return &copySingleImageResult{
 		manifest:         destManifest,
 		manifestMIMEType: destManifestType,
 		manifestDigest:   srcManifestDigest,
+		compressionAlgorithms: removeDuplicateCompression(compressionAlgos),
 	}, nil
 }
 
 // copyLayers copies layers from ic.src/ic.c.rawSource to dest, using and updating ic.manifestUpdates if necessary and ic.cannotModifyManifestReason == "".
-func (ic *imageCopier) copyLayers(ctx context.Context) error {
+func (ic *imageCopier) copyLayers(ctx context.Context) ([]compressiontypes.Algorithm, error) {
+	res := []compressiontypes.Algorithm{}
```

Comment:
Absolutely non-blocking: `res` might be too short a name for such a long function`. `compressionAlgos`? Or maybe define the value much later in the function, so that it has a short scope.